plugin/src/main/java/org/opensearch/ml/action/models/DeleteModelTransportAction.java

```diff
@@ -333,6 +478,59 @@ private Boolean isModelNotDeployed(MLModelState mlModelState) {
             && !mlModelState.equals(MLModelState.PARTIALLY_DEPLOYED);
     }
 
+    private <T> List<String> findDependentPipelines(
+        List<T> pipelineConfigurations,
+        String candidateModelId,
+        Function<T, Map<String, Object>> getConfigFunction,
+        Function<T, String> getIdFunction
+    ) {
+        List<String> dependentPipelineConfigurations = new ArrayList<>();
+        for (T pipelineConfiguration : pipelineConfigurations) {
+            Map<String, Object> config = getConfigFunction.apply(pipelineConfiguration);
+            if (searchThroughConfig(config, candidateModelId, "")) {
+                dependentPipelineConfigurations.add(getIdFunction.apply(pipelineConfiguration));
+            }
+        }
+        return dependentPipelineConfigurations;
+    }
+
+    // This method is to go through the pipeline configs and only when the key is model id and value is
+    // 1. String and equal to candidate id 2. A list of String containing candidate id We will return True. Otherwise False
+    private Boolean searchThroughConfig(Object searchCandidate, String candidateId, String targetModelKey) {
```

Comment:
targetModelKey is always an empty string. I do not think there is any need for this parameter